libsolidity/interface/CompilerStack.cpp

```diff
@@ -927,6 +928,12 @@ map<string, unsigned> CompilerStack::sourceIndices() const
 	return indices;
 }
 
+langutil::CharStream const* CompilerStack::charStreamForSourceIndex(unsigned _sourceIndex) const
+{
+	solAssert(_sourceIndex < m_sources.size(), "Cannot get CharStream of higher index than sources available.");
```

Comment:
This relies on the way `sourceIndices()` works without assuring it. It also ignores the `yulUtilityFileName`. Can you use a common data source for both functions? It does not have to be fast, the number of source files is less than 100 I would say.